WalletWasabi.Gui/Controls/WalletExplorer/TransactionViewModel.cs

```diff
@@ -28,8 +31,25 @@ public TransactionViewModel(TransactionInfo model)
 			ClipboardNotificationOpacity = 0;
 
 			CopyTransactionId = ReactiveCommand.CreateFromTask(TryCopyTxIdToClipboardAsync);
+			
+			OpenTransactionInfo = ReactiveCommand.Create(() =>
+			{
+				var shell = IoC.Get<IShell>();
+
+				var transactionInfo = shell.Documents?.OfType<TransactionInfoTabViewModel>()?.FirstOrDefault(x => x.Transaction?.TransactionId == TransactionId);
 
-			CopyTransactionId.ThrownExceptions
+				if (transactionInfo is null)
+				{
+					transactionInfo = new TransactionInfoTabViewModel(this);
```

Comment:
It is better to use a business object here IMO, like `TransactionInfo`. What if I just want to show the information about any transaction, now I need to generate a `TransactionViewModel`. What the `TransactionInfoTabViewModel` really uses is `TransactionInfo`.

Reply:
Yeah i think that makes sense, will do that instead of linking TransactionViewModel itself